Ranking Criteria 2022 (#166)
Quaver/Quaver.Wiki: en-US/ranking/criteria.md

- Unofficial cut versions of songs must use `(Cut Ver.)` at the end of the title. If the title contains a length marker by default, it should be replaced with `(Cut Ver.)`. This rule does not apply to songs that are shortened in ways that nearly match their original versions or songs that consist of a full loop from a looping track.
- If a song has been edited to have higher tempo, `(Sped Up Ver.)` marker must be used at the end of the title. If the title contains similar marker by default, it should be replaced with `(Sped Up Ver.)`.
- If a song has been edited to have lower tempo, `(Slowed Down Ver.)` marker must be used at the end of the title. If the title contains similar marker by default, it should be replaced with `(Slowed Down Ver.)`.
- Special unicode characters must be filtered to their nearest standard equivalent. Stars such as `★ ☆` should be substituted with an asterisk (`*`). Other special characters should be romanized or left out on a case-by-case basis.

# Media

## Rules

- **The total file size of the mapset must not exceed 50 MB.**

## Background

### Rules

> - **Every map in the mapset must have a background image.** Using different background images for different maps is acceptable. A mapset does not require a banner image.

Reviewer: Move the "A mapset does not require a banner image." to guidelines as "While not required, it is encouraged to include a banner image for your mapset."

- **The resolution of a background image must be at least 1280x720.**
- **The file size of a background image must not exceed 4 MB.**
- **The resolution of a banner image must be at least 421x82, if included.** This resolution also represents the default size and aspect ratio.
- **The background must be appropriate for players of 13+ years old and should not contain explicit content.**

### Guidelines

- The background should show relevance to the song in some way. While mappers are allowed creative freedom when it comes to choosing backgrounds, completely unrelated background images that do not compliment the song in any way should generally not be used.

## Audio

### Rules

> - **The mapset must contain only one audio file.**  Mapsets with multiple audio files are not eligible to be ranked.

Reviewer: Make an exception for keysound files if we care about that

- **MP3 is the only allowed audio file format.**
- **The maximum bitrate allowed for MP3 files is 192kbps.**

> - **The audio must be appropriate for players of all ages and should not include content that is offensive just for the sake of being offensive.**

Reviewer: implement explicit tag?

Author: requires backend changes, skip for now

# Maps

## Rules

The following requirements are for the individual maps themselves.

- **Autoplay must be able to achieve a 100% (X-Grade) score.**
- **The maps must not have overlapping notes.**

> - **The maps cannot have more than 30 seconds of consecutive break time.**

Reviewer: It seems a majority of us are fine with this being a guideline instead, something like "Avoid having long breaks in your map"

- **More than 75% of the length of the song must have notes to play.**
- **At least one note must be placed in every column.**
- **The maps must be at least 45 seconds long.**
- **The maps must be timed accurately.** The waveform in the editor may be used for more accurate timing.
- **The maps must have a song select preview point that sufficiently compliments the song.** Preview point is commonly placed at the beginning of a clear segment in the song, such as the chorus. A solid preview point attracts more players to try the map.

### Guidelines

> - **All maps in a mapset should use the same timing points.**

Reviewer: Clarify that this is mostly for timing line art or some other jank.

- **Notes should correlate to sounds in the music.** This includes but is not limited to:
     - Representing distinct sounds with certain amounts of notes
     - Representing distinct sounds with certain patterns of notes
- **Using layering is encouraged.** Mapping different instruments in the song will make the map reflect the music more.
- **Paying attention to chord/pattern consistency is encouraged.** Mapping certain elements of the song with very different ideas in quick succession without justification lowers the map's quality.
- **The intensity of the map should generally be synced with the music.** If a song is relatively mellow, the map should reflect this and vice versa.

> - Examples of cases where **simplifications to completely accurately following the music should be favoured** include but are not limited to:

Reviewer: ```suggestion
- Cases where **simplification of patterns may be preferred over following the music completely accurately** include but are not limited to:
```

>      - Causing an inappropriate difficulty spike relative to the rest of the map

Reviewer: ```suggestion
     - When an inappropriate difficulty spike would otherwise be made
```

| ------------- | :-------------------: |
| **0:45-2:29** |           2           |
| **2:30+**     |           1           |

### Rules

- The standard names for difficulties in a mapset are **Beginner, Easy, Normal, Hard, Insane, and Expert**. Should you choose not to use the standard names, the difficulty names in your set must show clear sign of progression in difficulty consistent with that of the mapset and should be easy to interpret. The highest difficulty of each game mode is always free to have a custom difficulty name.
     - Standard difficulty name may be added to the end of custom difficulty name in brackets to improve the interpretation of difficulty.
- You are **not** permitted to skip difficulties for **sets with a song length between 0:45-2:29.** For instance, if you are making a 1 minute set with an Insane, you can either create a Hard or an Expert difficulty. Having a Beginner and an Insane is not allowed. **This rule does not apply to sets with a song length of 2:30 or longer.**
- **Each game mode is treated separately.** If you are creating a mapset for both 4 Keys and 7 Keys, you must have a difficulty spread for both game modes.

> - **If a mapset has two game modes, each difficulty must be prefaced with either `4K` or `7K`** (Example: `7K Insane`).

Reviewer: Clarify that guest mappers' names can come before this

> - **If a mapset contains guest difficulties, the guest difficulties creators must be credited in their respective difficulty names** (Example: `Flamingo's Hard`). Mapset's host should not be credited in this fashion.

Reviewer: I think there was some discussion before about loosening this rule. For mapsets that are more like a collab between mappers, allowing the host's name to appear in the diff name makes it look more natural:
- Guest 1's Easy
- Guest 2's Normal
- Host's Hard
- Guest 3's Insane

> - **The number of difficulties created by the host of a mapset must be equal or higher than the largest number of guest difficulties created by a single user.** If a mapper with guest difficulties has contributed to the set with more difficulties than the host, the host should be changed.

Reviewer: I remember some discussion before about loosening this rule as well.

One might argue that for a spread where a guest maps easy, normal, hard and host maps insane, expert, it would make more sense for the host to stay the same
